### PR TITLE
Merge store data feature into main extension branch

### PR DESCRIPTION
**Pull request by falbor45**

### CODE REVIEW

File: extension/popup.js [hunk 1]
```diff
@@ -1,26 +1,57 @@
...
 var peselInput = document.getElementById('PeselCalc')
 var surnameInput = document.getElementById('SurnameCalc')
 var zipcodeInput = document.getElementById('ZipCodeCalc')
 var licenseInput = document.getElementById('licenseDate')
-function updateInput(event) {
-    NameInput.setAttribute("value", event.target.value)
+
+function updateNameInput(event) {
+    nameInput.setAttribute("value", event.target.value)
+    localStorage.setItem('name', event.target.value)
```
Comment (MKwestorowski) [inline extension/popup.js]: I think there start a problem. If somebody wrote incorrect data, that gonna be incorrect at another places. May we should try to validate before set to storage. What do u think?

File: extension/popup.js [hunk 2]
```diff
@@ -35,4 +66,35 @@ var testF = function() {
...
+    surnameInput.value = surname
+    zipcodeInput.value = zipcode
+    licenseInput.value = license
+}
+
+function clearAllInputs() {
+    var selectAllInputs = document.getElementsByTagName('input')
+
+    for (i = 0; i < selectAllInputs.length; i++) {
+        if (selectAllInputs[i].type == 'text' || selectAllInputs[i].type == 'number') {
```
Comment (MKwestorowski) [inline extension/popup.js:99]: Why do You here using two equals signs? Another future is what type means? I looked for some mdm documentation but got only .type() in jQuery. If u mean typeof there isn't thing like text. I hope u mean string 💃 . Have fun!

**Review by MKwestorowski — CHANGES_REQUESTED** [review]

File: extension/popup.js (hunk 2 above)
Comment (falbor45) [reply extension/popup.js:99]: You are absolutely right. ".type" is not a function that exists solely in bounds of Javascript itself. ".type" is a Web API function and is the most efficient way to find out control form's type I have found so far. If you try to console.log any of the iterrated elements, you will see either 'text', 'number', or 'date' as an output. :) 
TLDR;
'.type' is all good, and AFAIK the most efficient way, it is not related to jQuery at all (in this very situation)

https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/type

File: extension/popup.js (hunk 1 above)
Comment (Wojtekkas) [reply extension/popup.js]: We can validate inputs just by using simple patterns like=[A-Za-z] but we cant prevent users from making typo in name.

File: extension/popup.js [hunk 3]
```diff
@@ -35,4 +66,35 @@ var testF = function() {
...
+    nameInput.value = name
+    kilometersInput.value = km
+    peselInput.value = pesel
+    surnameInput.value = surname
+    zipcodeInput.value = zipcode
+    licenseInput.value = license
+}
+
+function clearAllInputs() {
+    var selectAllInputs = document.getElementsByTagName('input')
```
Comment (Wojtekkas) [inline extension/popup.js:96]: We need to be careful here. We got permission to all tabs so we will erase all user inputs and we dont want to do it.

**Review by Wojtekkas — APPROVED** [review]
Looks okay to me.

Just think about the line 86.

File: extension/popup.js (hunk 3 above)
Comment (falbor45) [reply extension/popup.js:96]: It will clean only the inputs inside the extension as it is treated as a completely separate document.

**Review by MKwestorowski — APPROVED** [review]

File: extension/popup.js [hunk 4]
```diff
@@ -14,6 +14,14 @@ document.addEventListener('DOMContentLoaded',function() {
...
 
 var testF = function() {
+
+    sessionStorage.setItem('name', NameInput.value)
+    sessionStorage.setItem('km', kilometersInput.value)
+    sessionStorage.setItem('pesel', peselInput.value)
+    sessionStorage.setItem('surname', surnameInput.value)
+    sessionStorage.setItem('zipcode', zipcodeInput.value)
+    sessionStorage.setItem('license', licenseInput.value)
+
```
Comment (RafalPaluch) [inline extension/popup.js]: This is the good place for refactoring. Now its Ok, but for sure the will be more lines like this in this section.

File: extension/popup.js (hunk 1 above)
Comment (RafalPaluch) [reply extension/popup.js]: For sure,we need validation for  Zip Code etc. to prevent use O instead of 0 etc

**Review by RafalPaluch — APPROVED** [review]
Ok

**Review by RafalPaluch — APPROVED** [review]
